LibreNMS/Data/Store/Kafka.php

```diff
@@ -0,0 +1,326 @@
+            $excluded_device_fields = Config::get('kafka.device-fields-exclude'); // comman separated string
+            $excluded_device_fields_arr = [];
+
+            if ($excluded_groups != null && strlen($excluded_groups) > 0) {
+                // convert into array
+                $excluded_groups_arr = explode(',', strtoupper($excluded_groups));
+
+                $device_groups = $device_data->groups;
+                foreach ($device_groups as $group) {
+                    // The group name will always be parsed as lowercase, even when uppercase in the GUI.
+                    if (in_array(strtoupper($group->name), $excluded_groups_arr)) {
+                        Log::debug('KAFKA: Skipped parsing to Kafka, device is in group: ' . $group->name);
+
+                        return;
+                    }
+                }
+            }
+
+            if ($excluded_measurement != null && strlen($excluded_measurement) > 0) {
+                // convert into array
+                $excluded_measurement_arr = explode(',', $excluded_measurement);
+
+                if (in_array($measurement, $excluded_measurement_arr)) {
+                    Log::debug('KAFKA: Skipped parsing to Kafka, measurement is in measurement-excluded: ' . $measurement);
+
+                    return;
+                }
+            }
+
+            if ($excluded_device_fields != null && strlen($excluded_device_fields) > 0) {
+                // convert into array
+                $excluded_device_fields_arr = explode(',', $excluded_device_fields);
+            }
+
+            // start
+            $stat = Measurement::start('write');
+
+            $tmp_fields = [];
+            $tmp_tags = [];
+            $tmp_tags['device_groups'] = implode('|', $device_data->groups->pluck('name')->toArray());
+
+            foreach ($tags as $k => $v) {
+                if (empty($v)) {
+                    $v = '_blank_';
+                }
+                $tmp_tags[$k] = $v;
+            }
+            foreach ($fields as $k => $v) {
+                if ($k == 'time') {
+                    $k = 'rtime';
+                }
+
+                if (($value = $this->forceType($v)) !== null) {
+                    $tmp_fields[$k] = $value;
+                }
+            }
+
+            if (empty($tmp_fields)) {
+                Log::warning('KAFKA: All fields empty, skipping update', [
+                    'orig_fields' => $fields,
+                    'device_id' => $this->device_id,
+                ]);
+
+                return;
+            }
+
+            // create and organize data
+            $filteredDeviceData = array_diff_key($device->toArray(), array_flip($excluded_device_fields_arr));
+            // add current time to the data
+            $filteredDeviceData['current_polled_time'] = Carbon::now();
+
+            $resultArr = [
+                'measurement' => $measurement,
+                'device' => $filteredDeviceData,
```

Comment:
should not be including the device database entry in here (as discussed on discord, duplicating here so we don't forget)

tags should include "hostname" which will be enough to fetch data from the API.